File: src/leetCodeProblems/BitsManupulation/SingleNumber136.java
package leetCodeProblems.BitsManupulation;

/**
 * LeetCode - https://leetcode.com/problems/single-number/
 *
 * Time Complexity - O(n)
 * Space Complexity - O(1)
 */
public class SingleNumber136 {

    public int singleNumber(int[] nums) {

        int ansXOR = 0;

        for (int i=0; i < nums.length; i++) {
            //System.out.println("Before ->" + ansXOR);
            ansXOR = ansXOR ^ nums[i];
            //System.out.println("After ->" + ansXOR);
        }

        return ansXOR;
    }

    public static void main(String[] args) {

        SingleNumber136 obj = new SingleNumber136();

        int[] nums = {4,1,2,1,2};
        System.out.println(obj.singleNumber(nums));
    }
}

File: src/leetCodeProblems/BitsManupulation/SingleNumber136.go
package main

/*
- Leetcode - https://leetcode.com/problems/single-number/
- TimeComplexity - O(n)
- Space Complexity - O(1)
*/

func singleNumber1(nums []int) int {
	ansXOR := 0

	for _, v := range nums {
		ansXOR ^= v
	}

	return ansXOR

}

// func main() {
// 	//nums := []int{4, 1, 2, 1, 2}

// 	//nums := []int{0, 1, 0, 1, 0, 1, 99}
// 	nums := []int{2, 2, 3, 2}
// 	log.Println(singleNumber1(nums))
// }
